### PR TITLE
Fix ams file manager

### PR DESCRIPTION
**Pull request by xTEddie**
- Adding checks to prevent methods to be called when properties with attachment are not present

### CODE REVIEW

File: __tests__/external/ACSAdapter/AMSFileManager.spec.ts [hunk 1]
```diff
@@ -57,20 +61,25 @@ describe('AMSFileManager', () => {
 
         const response = fileManager.getFileIds(metadata);
         expect(response).toStrictEqual(amsReferences);
+        expect(logger.startScenario).toBeCalledTimes(1);
     });
 
-    it('AMSFileManager.getFileIds() should return a nothing if invalid', async () => {
+    it('AMSFileManager.getFileIds() should return nothing if invalid', async () => {
```
Comment (elopezanaya) [inline __tests__/external/ACSAdapter/AMSFileManager.spec.ts:67]: is this valid?, if something is invalid, isnt better to return an error code?

File: __tests__/external/ACSAdapter/AMSFileManager.spec.ts [hunk 2]
```diff
@@ -99,7 +108,7 @@ describe('AMSFileManager', () => {
         expect(response).toStrictEqual(amsMetadata);
     });
 
-    it('AMSFileManager.getFileMetadata() should return a nothing if invalid', async () => {
+    it('AMSFileManager.getFileMetadata() should return nothing if invalid', async () => {
```
Comment (elopezanaya) [inline __tests__/external/ACSAdapter/AMSFileManager.spec.ts:111]: same question  about error code with invalid scenarios

File: src/external/ACSAdapter/AMSFileManager.ts [hunk 3]
```diff
@@ -81,6 +85,10 @@ class AMSFileManager {
     }
 
     public createFileIdProperty(fileIds: string[]): Record<string, string> | undefined {
+        if (!fileIds) {
```
Comment (elopezanaya) [inline src/external/ACSAdapter/AMSFileManager.ts:88]: I'm on the side of sending error code in this scenarios,

**Review by elopezanaya — CHANGES_REQUESTED** [review]

File: __tests__/external/ACSAdapter/AMSFileManager.spec.ts (hunk 1 above)
Comment (xTEddie) [reply __tests__/external/ACSAdapter/AMSFileManager.spec.ts:67]: It's valid. If there is no metadata, it should be a no-op. If there's `fileIds`, it means that there's an attachment then it should be parsed. If the parsing fails, it should only be logged then continue. We always have to ensure the FileManager does not throw any exception to ensure the conversation is flowing. If an exception was thrown then not caught, WebChat will stop working.

File: __tests__/external/ACSAdapter/AMSFileManager.spec.ts (hunk 2 above)
Comment (xTEddie) [reply __tests__/external/ACSAdapter/AMSFileManager.spec.ts:111]: It's a valid scenario as explained previously

**Pull request merged**